src/pretix/base/models/items.py

```diff
@@ -303,6 +303,48 @@ def filter_available(qs, channel='web', voucher=None, allow_addons=False):
     return qs.filter(q)
 
 
+def filter_variations_available(qs, channel='web', voucher=None, allow_addons=False):
+    # Channel can currently be a SalesChannel or a str, since we need that compatibility, but a SalesChannel
+    # makes the query SIGNIFICANTLY faster
+    from .organizer import SalesChannel
+
+    assert isinstance(channel, (SalesChannel, str))
+    q = (
+        Q(active=True)
+        & Q(Q(available_from__isnull=True) | Q(available_from__lte=time_machine_now()))
+        & Q(Q(available_until__isnull=True) | Q(available_until__gte=time_machine_now()))
+        & Q(item__active=True)
+        & Q(Q(item__available_from__isnull=True) | Q(item__available_from__lte=time_machine_now()))
+        & Q(Q(item__available_until__isnull=True) | Q(item__available_until__gte=time_machine_now()))
+        & Q(Q(item__category__isnull=True) | Q(item__category__is_addon=False))
+        & Q(item__require_bundling=False)
+    )
+
+    if isinstance(channel, str):
+        q &= Q(Q(all_sales_channels=True) | Q(limit_sales_channels__identifier=channel))
+        q &= Q(Q(item__all_sales_channels=True) | Q(item__limit_sales_channels__identifier=channel))
+    else:
+        q &= Q(Q(all_sales_channels=True) | Q(limit_sales_channels=channel))
+        q &= Q(Q(item__all_sales_channels=True) | Q(item__limit_sales_channels=channel))
+
+    if not allow_addons:
+        q &= Q(Q(item__category__isnull=True) | Q(item__category__is_addon=False))
```

Comment:
These two lines seem new, but what is the difference to line 319 above, which is always added to the queryset? Not sure, this here has any effect/makes a difference.